src/imageops/sample.rs

```diff
@@ -105,6 +105,20 @@ pub(crate) struct Filter<'a> {
     pub(crate) support: f32,
 }
 
+struct FloatNearest(f32);
+
+impl ToPrimitive for FloatNearest {
+    fn to_i64(&self) -> Option<i64> {
+        NumCast::from(self.0.round())
+    }
+    fn to_u64(&self) -> Option<u64> {
+        NumCast::from(self.0.round())
+    }
+    fn to_f64(&self) -> Option<f64> {
+        NumCast::from(self.0)
+    }
+}
```

Comment:
Very interesting solution :slightly_smiling_face: If I read it correctly then `to_f32` dispatches to `to_f64` hence no additional override is necessary. Maybe add this as a comment as it's not obvious at all that nothing was accidentally omitted.